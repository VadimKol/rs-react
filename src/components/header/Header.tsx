'use client';

import { type ReactNode } from 'react';

import { useTheme } from '@/hooks/useTheme';

import styles from './styles.module.scss';

export function Header(): ReactNode {
  const { theme, toggleTheme } = useTheme();

  return (
    <header className={styles.header}>
      <div className={styles.container} />
      <button
        className={`${styles.theme} ${theme === 'dark' ? styles.button_dark : styles.button_light}`}
        type="button"
        aria-label="Theme switcher"
        onClick={toggleTheme}
      />
    </header>
  );
}
