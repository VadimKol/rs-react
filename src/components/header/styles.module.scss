.header {
  position: relative;
  width: 100%;
  background-image: linear-gradient(to top, #393838, #000000);
}

.container {
  width: 300px;
  height: 160px;
  margin: auto;
  background-image: url("/src/assets/images/header.png");
  background-size: cover;
}
