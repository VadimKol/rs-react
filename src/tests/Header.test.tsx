import { render } from '@testing-library/react';

import { Header } from '@/components/header/Header';

describe('Header Component', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
