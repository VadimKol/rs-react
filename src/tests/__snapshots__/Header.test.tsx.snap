// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > renders correctly 1`] = `
<div>
  <header
    class="_header_039d8e"
  >
    <div
      class="_container_039d8e"
    />
    <button
      aria-label="Theme switcher"
      class="_theme_039d8e _button_dark_039d8e"
      type="button"
    />
  </header>
</div>
`;
